src/PyMca5/PyMcaGui/io/hdf5/Hdf5NodeView.py

```diff
@@ -94,36 +94,24 @@ class Plot1DViewWithPlugins(DataViews._Plot1dView):
     def createWidget(self, parent):
         return Plot1DWithPlugins(parent=parent)
 
-
-class Plot2DWithPlugins(Plot2D):
-    """Add a plugin toolbutton to a Plot2D"""
-    def __init__(self, parent=None):
-        Plot2D.__init__(self, parent)
-
-        self._toolbar = qt.QToolBar(self)
-        self.addToolBar(self._toolbar)
-        pluginsToolButton = PluginsToolButton(plot=self, parent=self,
+class Plot2DViewWithPlugins(DataViews._Plot2dView):
+    def createWidget(self, parent):
+        widget = super().createWidget(parent)
+        widget.setKeepDataAspectRatio(False)
+        widget.pymcaToolbar = qt.QToolBar(widget)
+        widget.addToolBar(widget.pymcaToolbar)
+        pluginsToolButton = PluginsToolButton(plot=widget, parent=widget,
                                               method="getPlugin2DInstance")
 
         if PLUGINS_DIR:
             pluginsToolButton.getPlugins(
                     method="getPlugin2DInstance",
                     directoryList=PLUGINS_DIR)
-        self._toolbar.addWidget(pluginsToolButton)
-        if hasattr(self, "getIntensityHistogramAction"):
-            self.getIntensityHistogramAction().setVisible(True)
+        widget.pymcaToolbar.addWidget(pluginsToolButton)
+        if hasattr(widget, "getIntensityHistogramAction"):
+            widget.getIntensityHistogramAction().setVisible(True)
```

Comment:
This is there since 9 years, safe to use IMO:

```suggestion
        widget.getIntensityHistogramAction().setVisible(True)
```